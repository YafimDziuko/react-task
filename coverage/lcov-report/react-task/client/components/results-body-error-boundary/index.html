<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for react-task/client/components/results-body-error-boundary</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> react-task/client/components/results-body-error-boundary
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">64.86% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>24/37</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">35.48% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>11/31</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/12</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="results-body-error-boundary.jsx"><a href="results-body-error-boundary.jsx.html">results-body-error-boundary.jsx</a></td>
	<td data-value="64.86" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 64%;"></div><div class="cover-empty" style="width:36%;"></div></div></td>
	<td data-value="64.86" class="pct medium">64.86%</td>
	<td data-value="37" class="abs medium">24/37</td>
	<td data-value="35.48" class="pct low">35.48%</td>
	<td data-value="31" class="abs low">11/31</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="12" class="abs medium">6/12</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri May 04 2018 16:45:48 GMT+0300 (Belarus Standard Time)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
<script src="..\..\..\..\block-navigation.js"></script>
</body>
</html>
