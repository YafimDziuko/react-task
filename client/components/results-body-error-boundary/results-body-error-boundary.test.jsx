import React from 'react';
import { ResultsBodyErrorBoundary } from './results-body-error-boundary';

describe('<ResultsBodyErrorBoundary/>', function() {
  it('should render ResultsBodyErrorBoundary and match snapshot', function() {
      const wrapper = shallow(<ResultsBodyErrorBoundary children={<div>Hello</div>} />);

      expect(wrapper).toMatchSnapshot();
  });

  it('should render ResultsBodyErrorBoundary with Full Film Item', function() {
    const wrapper = shallow(<ResultsBodyErrorBoundary children={<div>Hello</div>}/>);
    wrapper.setState({hasError: true});

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().hasError).toEqual(true);
  });
});
