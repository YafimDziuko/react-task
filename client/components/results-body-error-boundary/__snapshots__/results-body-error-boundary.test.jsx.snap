// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsBodyErrorBoundary/> should render ResultsBodyErrorBoundary and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsBodyErrorBoundary
    fullItem={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ResultsBodyErrorBoundary/> should render ResultsBodyErrorBoundary with Full Film Item 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsBodyErrorBoundary />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Something went wrong.",
      "className": "error-boundary",
    },
    "ref": null,
    "rendered": "Something went wrong.",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Something went wrong.",
        "className": "error-boundary",
      },
      "ref": null,
      "rendered": "Something went wrong.",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
